lib/elixir/src/elixir_exp.erl

```diff
@@ -329,6 +329,18 @@ expand({Atom, Meta, Args}, E) when is_atom(Atom), is_list(Meta), is_list(Args) -
 
 %% Remote calls
 
+expand({{'.', Meta, [erlang, 'orelse']}, _, [Left, Right]}, #{context := nil} = Env) ->
+  Generated = ?generated(Meta),
+  TrueClause = {'->', Generated, [[true], true]},
+  FalseClause = {'->', Generated, [[false], Right]},
+  expand_boolean_check(Left, TrueClause, FalseClause, Meta, Env);
+
+expand({{'.', Meta, [erlang, 'andalso']}, _, [Left, Right]}, #{context := nil} = Env) ->
```

Comment:
Actually, nevermind. This approach is fine because the context is either guard, match or nil and we can't have it in matches anyway. :)